### PR TITLE
GH-3127: Add ResultsDirectory to VSTestSettings. (see #3127)

### LINKED ISSUE #3127: Add ResultsDirectory to VSTestSettings
ITNOA

I think its sound good to add ResultsDirectory to VSTestSettings.

### PR DESCRIPTION
**Pull request by soroshsabz**
This PR resolved #3127

### CODE REVIEW

**Comment by soroshsabz** [comment]
@devlead Please review it :)

thanks

**Review by augustoproiete — CHANGES_REQUESTED** [review]
Thanks @soroshsabz. LGTM.

Could you please squash the commits and rebase against latest `develop`?

**Comment by soroshsabz** [comment]
@augustoproiete I done it.

File: src/Cake.Common/Tools/VSTest/VSTestRunner.cs [hunk 1]
```diff
@@ -117,6 +117,11 @@ private ProcessArgumentBuilder GetArguments(IEnumerable<FilePath> assemblyPaths,
                 builder.AppendSwitchQuoted("/Diag", ":", settings.Diag.MakeAbsolute(_environment).FullPath);
             }
 
+            if (settings.ResultsDirectory != null)
+            {
+                builder.AppendSwitchQuoted("/ResultsDirectory", ":", settings.ResultsDirectory.MakeAbsolute(_environment).FullPath);
```
Comment (augustoproiete) [inline src/Cake.Common/Tools/VSTest/VSTestRunner.cs:122]: :+1: Looks good / prior art

File: src/Cake.Common/Tools/VSTest/VSTestSettings.cs [hunk 2]
```diff
@@ -73,6 +73,13 @@ public sealed class VSTestSettings : ToolSettings
         /// </summary>
         public FilePath Diag { get; set; }
 
+        /// <summary>
+        /// Gets or sets the result directory.
+        /// Test results directory will be created in specified path if not exists.
+        /// VSTest.Console.exe flag <see href="https://docs.microsoft.com/en-us/visualstudio/test/vstest-console-options#ResultDirectory">/ResultsDirectory</see>.
+        /// </summary>
+        public DirectoryPath ResultsDirectory { get; set; }
```
Comment (augustoproiete) [inline src/Cake.Common/Tools/VSTest/VSTestSettings.cs:81]: :+1: Matches command-line argument in VSTest-Console [docs](https://docs.microsoft.com/en-us/visualstudio/test/vstest-console-options#ResultDirectory)

[image](https://docs.microsoft.com/en-us/visualstudio/test/vstest-console-options#ResultDirectory)

**Review by augustoproiete — APPROVED** [review]
LGTM

**Comment by augustoproiete** [comment]
@soroshsabz your changes have been merged, thanks for your contribution :+1:

**Pull request merged**